### ISSUE TITLE
Cannot replicate reported PIE results with provided code & config

### ISSUE THREAD

**Issue by aeskandari68** [OSU-Haolin/Pedestrian_Crossing_Intention_Prediction#20]
Hi, thank you for sharing your work and releasing the code.

I'm currently trying to replicate your results on the PIE dataset using the `MASK_PCPA_4_2D` model. However, I haven’t been able to match the reported performance.

# My setup:
- I used the provided codebase and configuration.

- I also downloaded your PIE.zip and verified that my config file matches what you included.

- I uncommented the validation dataset loading line to ensure that validation was used during training.

- I trained with and without validation for up to 200 epochs.

# Problem:
When using a validation set, the model appears to overfit significantly — validation loss increases while training metrics keep improving.
When training without validation, the final test results are still substantially worse than those reported in your paper.

My config file:
```yaml
data_opts:
  data_split_type: default
  fstride: 1
  min_track_size: 76
  sample_type: all
  seq_type: crossing
  subset: default

model_opts:
  apply_class_weights: true
  balance_data: false
  dataset: pie
  enlarge_ratio: 1.5
  fusion_method: sum
  fusion_point: early
  generator: true
  model: MASK_PCPA_4_2D
  normalize_boxes: true
  obs_input_type:
    - local_context_cnn
    - mask_cnn
    - pose
    - box
    - speed
  obs_length: 16
  overlap: 0.6
  time_to_event: [30, 60]

net_opts:
  backbone: vgg
  cell_type: gru
  global_pooling: avg
  num_hidden_units: 256
  regularizer_val: 0.0001

train_opts:
  batch_size: 2
  epochs: 200
  learning_scheduler: null
  lr: 5.0e-05
```
My enviroment
```bash
$ pip list | grep ten
tensorboard            2.2.0
tensorboard-plugin-wit 1.6.0
tensorflow             2.2.0
tensorflow-estimator   2.2.0
tensorflow-gpu         2.2.0
```

# Results after 200 epochs (no validation):
```yaml
results:
  acc: 0.8404088050314465
  auc: 0.8170166945792534
  f1: 0.7292129835482436
  precision: 0.6978723404255319
  recall: 0.7635009310986964
```
# Learning Curves:

![Image](https://github.com/user-attachments/assets/e7d04027-45e8-4f98-b006-244035d3ea52)

## Training command
```bash
python train_test.py -c config_files_pie/ours/MASK_PCPA_jaad_2d.yaml
```
Other plots:
![Image](https://github.com/user-attachments/assets/5131333b-f6f7-4736-9097-cb9b97ab6cf4)

![Image](https://github.com/user-attachments/assets/26d6b63d-9559-4af8-9d67-c949edd9d16b)

# Questions:
1. Did you use a validation set during training for the final reported PIE results?

2. Were there any specific tricks, learning rate schedules, or seeds used that might not be reflected in the public config/code?

Thank you again for your time and contribution — and for any insight you can provide.
